### PR TITLE
feat: add polkadot

### PR DESCRIPTION
**Pull request by zachferland**

### CODE REVIEW

**Comment by zachferland** [comment]
Problem now is sr25519 signatures are apparently not deterministic, so fine for linking, but can not be used to auth as we do. Not sure how we can resolve this yet.

File: src/blockchains/__tests__/polkadot.test.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import { waitReady } from '@polkadot/wasm-crypto';
+
+const did = 'did:3:bafysdfwefwe'
+const net = 'polkadot:b0a8d493285c2df73290dfb7e61f870f'
+const seed = hexToU8a('0xabf8e00000000000000000000000000000000000000000000000000000000000')
+
+// primary and supported by polkadot extension
+const keyringSr25519 = createTestKeyring({ type: 'sr25519' })
+const keyringEd25519 = createTestKeyring({ type: 'ed25519' })
+const keyringSecp256k = createTestKeyring({ type: 'ecdsa' })
```
Comment (oed) [inline src/blockchains/__tests__/polkadot.test.ts:18]:
```suggestion
const keyringSecp256k1 = createTestKeyring({ type: 'ecdsa' })
```

File: src/blockchains/polkadot.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        message: linkMessageHex,
+        signature: res.signature,
+        account: account.toString()
+    }
+    return proof
+}
+
+async function authenticate(message: string, account: AccountID, signer: any): Promise<string> {
+    const res = await signer.signRaw({address: account.address, data: stringHex(message)})
+    return res.signature
```
Comment (oed) [inline src/blockchains/polkadot.ts]: Probably a good idea to take the hash of the signature.

**Review by oed — COMMENTED** [review]
Looks good overall.

That's strange about the deterministic signatures. Can you ask Steve from Parity about it to double check? Maybe there is some other way to do it?

File: src/blockchains/polkadot.ts (hunk 2 above)
Comment (zachferland) [reply src/blockchains/polkadot.ts]: ah yes, definitely

File: src/blockchains/__tests__/polkadot.test.ts (hunk 1 above)
Comment (oed) [reply src/blockchains/__tests__/polkadot.test.ts:18]: Noticed that the name is incorrect in various places.

**Pull request merged**